### ISSUE TITLE
Certain MP3 files don't keep looping when using sf::Music

### ISSUE THREAD

**Issue by Biohazard90** [SFML/SFML#2215]
## Certain MP3 files not looping

Certain mp3 files are not looping correctly when playing them with sf::Music and looping enabled. One of them doesn't loop at all (issue 1), the other stops looping after the first time (issue 2).

## Your environment

* Windows 10 Pro
* SFML 2.6.X
* VS 17.3.3 Pro, MSVC C++ 20
* Static linking

## Steps to reproduce

Tell us how to reproduce this issue. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve), you can use the follow template as a start:

```cpp
#include "sfml/Audio/Music.hpp"

#include <Windows.h>

using namespace std;

int main(int argc, char *argv[])
{
	sf::Music music;
	music.setLoop(true);
	music.setVolume(10);

	// This file stops playing after the first playthrough (issue 1)
	//if (music.openFromFile("SIERRA - UNBROKEN (mp3cut.net).mp3"))
	//	music.play();

	// This file stops playing after the second playthrough (issue 2)
	if (music.openFromFile("Feeling - Tung Wasabi (mp3cut.net).mp3"))
		music.play();
	
	MSG msg = {};
	while (GetMessage(&msg, 0, 0, 0))
	{
		TranslateMessage(&msg);
		DispatchMessage(&msg);
	}

	return 0;
}
```

## Expected behavior

The music should never stop playing.

## Actual behavior

The file "SIERRA - UNBROKEN (mp3cut.net).mp3" stops playing after the first playthrough (issue 1), the file "Feeling - Tung Wasabi (mp3cut.net).mp3" stops playing after the second playthrough (issue 2).

## Analysis
In both cases the problem is that SFML only loops a file if the expected EOF point is reached and minimp3 doesn't reach it, instead it fails reading more data.

Regarding issue 1, I believe the file may report the incorrect number of samples and minimp3 takes it at face value, then just stops streaming too early when EOF is reached. Since I currently needed a solution to this, I made a workaround for myself so SFML continues looping when EOF is reached too early: https://github.com/Biohazard90/SFML/commit/bcd67988848fe76c0e498fdc8778fd38428b8612 Note, this is not a fix I am proposing, it's just to illustrate the issue and how I'm solving it since I cannot fix the mp3 files I'm working with.

Regarding issue 2, minimp3 seems to be incorrectly seeking to 0 and I proposed a fix here already: https://github.com/lieff/minimp3/issues/101

Here are the two mp3 files that have these issues:
[Example mp3 files](https://github.com/SFML/SFML/files/9658753/Release.zip)

**Comment by karplucas** [comment]
SFML supports the audio file formats WAV, OGG/Vorbis and FLAC. Due to licensing issues MP3 is not supported.

**Comment by Biohazard90** [comment]
> SFML supports the audio file formats WAV, OGG/Vorbis and FLAC. Due to licensing issues MP3 is not supported.

No, MP3 is supported by SFML 2.6.X. The last patents have expired in 2017.

**Comment by karplucas** [comment]
You're right, didnt know it was added to SFML 2.6.x.

Reproduced your issue and checked whether AL was processing the stream buffers correctly and it is indicating that there are none processed. So loop looks to be working fine, but likely the seek to Time::Zero once the mp3 is finished is not working.

Not knowledgeable enough to edit minimp3, but I did try to verify your solution and did not see it working for me.

**Comment by eXpl0it3r** [comment]
Looks like @lieff is currently not responding, as such, I don't see us fixing this for the up coming SFML 2.6 release and I'll move it to SFML 2.6.1 in hopes they come back at one point.

**Comment by TiEul** [comment]
Hey @eXpl0it3r, at this point it is kind of safe to assume that minimp3 is abandoned, is it maybe possible that the SFML team creates a fork of it and maintains that separate fork for future bug fixes? I am not sure what the best-practice is in such a case, but it is a bit of a shame that this cannot be fixed due to an abandoned repo.

**Comment by eXpl0it3r** [comment]
I've tested the provided files with the dr_mp3 implementation and they loop without issues. dr_mp3 uses most of the minimp3 implementation with some API improvements and is written and maintained by the same person that wrote [miniaudio](https://github.com/mackron/miniaudio) and they're still quite active.

It's been a while but if anyone here wants to test the #2995 PR, @Biohazard90 / @TiEul

**Comment by TiEul** [comment]
@eXpl0it3r Thanks, we don't have the resources to verify it right now but we'll add it to our to-do list and we'll provide feedback here.

**Comment by eXpl0it3r** [comment]
After the final release of dr_mp3, the provided MP3 failed to loop again, as the expected loop point at the end of the file is never reached.

The author of dr_mp3 checked the provided MP3s and their conclusion is, that the MP3s are likely corrupt in some way. They both seem to mention `mp3cut.net`, which might produce problematic MP3s.

I think, it's still an improvement to update to dr_mp3. Additionally, I think we should add some case in SFML for handling loop points and the end of the file, when the counts don't end up matching.

**Comment by Biohazard90** [comment]
Thanks for the heads up. Yes, the files are definitely "defective". As explained in my OP, I believe they report an incorrect number of total samples in the header.

I think it would be great if SFML would still officially support looping in this case since conceptually the expected behavior could still be fulfilled.
